setup.py

```diff
@@ -143,10 +151,173 @@ def _header_bad(self, path):
+        warnings.warn(
+            'Unable to determine Proj4 version. Ensure you have %s or later '
+            'installed, or installation may fail.' % (
+                '.'.join(str(v) for v in PROJ_MIN_VERSION), ))
+        proj_version = (0, 0, 0)
+
+    return proj_version
+
+
+conda = os.getenv('CONDA_DEFAULT_ENV')
+if conda is not None and conda in sys.prefix:
+    # Conda does not provide pkg-config compatibility, but the search paths
+    # should be set up so that nothing extra is required. We'll still check
+    # the version, though.
+    proj_version = find_proj_version_by_program(conda)
+
+    proj_includes = []
+    proj_libraries = ['proj']
+    proj_library_dirs = []
+
+else:
+    try:
+        proj_version = subprocess.check_output(['pkg-config', '--modversion',
+                                                'proj'],
+                                               stderr=subprocess.STDOUT)
+        proj_version = tuple(int(v) for v in proj_version.split(b'.'))
+        proj_includes = subprocess.check_output(['pkg-config', '--cflags',
+                                                 'proj'])
+        proj_clibs = subprocess.check_output(['pkg-config', '--libs', 'proj'])
+    except (OSError, ValueError, subprocess.CalledProcessError):
+        proj_version = find_proj_version_by_program()
+        if proj_version < PROJ_MIN_VERSION:
+            print(
+                'Proj4 version %s is installed, but cartopy requires at least '
+                'version %s.' % ('.'.join(str(v) for v in proj_version),
+                                 '.'.join(str(v) for v in PROJ_MIN_VERSION)),
+                file=sys.stderr)
+            exit(1)
+
+        proj_includes = []
+        proj_libraries = ['proj']
+        proj_library_dirs = []
+    else:
+        if proj_version < PROJ_MIN_VERSION:
+            print(
+                'Proj4 version %s is installed, but cartopy requires at least '
+                'version %s.' % ('.'.join(str(v) for v in proj_version),
+                                 '.'.join(str(v) for v in PROJ_MIN_VERSION)),
+                file=sys.stderr)
+            exit(1)
+
+        if PY3:
+            proj_includes = proj_includes.decode()
+            proj_clibs = proj_clibs.decode()
+
+        proj_includes = proj_includes.split()
+        proj_libraries = []
+        proj_library_dirs = []
+        for entry in proj_clibs.split():
+            if entry.startswith('-L'):
+                proj_library_dirs.append(entry[2:])
+            elif entry.startswith('-l'):
+                proj_libraries.append(entry[2:])
+
+# Python dependencies
+extras_require = {}
+for name in os.listdir(os.path.join(HERE, 'requirements')):
+    with open(os.path.join(HERE, 'requirements', name), 'r') as fh:
+        section, ext = os.path.splitext(name)
+        extras_require[section] = []
+        for line in fh:
+            if line.startswith('#'):
+                pass
+            elif line.startswith('-'):
```

Comment:
This character was a leftover of when I tried to put `-rdefault.txt` in the extended requirement files to also include the default requirements, but that flag did not work with `conda`.